src/repo/mod.rs

```diff
@@ -103,16 +102,31 @@ pub struct Repo<TRepoTypes: RepoTypes> {
     block_store: TRepoTypes::TBlockStore,
     data_store: TRepoTypes::TDataStore,
     events: Sender<RepoEvent>,
-    subscriptions: Mutex<SubscriptionRegistry<Cid, Block>>,
+    subscriptions: SubscriptionRegistry<Block>,
 }
 
 #[derive(Clone, Debug)]
 pub enum RepoEvent {
     WantBlock(Cid),
+    UnwantBlock(Cid),
     ProvideBlock(Cid),
     UnprovideBlock(Cid),
 }
 
+impl From<Request> for RepoEvent {
+    fn from(req: Request) -> Self {
+        if let Request {
+            kind: RequestKind::GetBlock(cid),
+            ..
+        } = req
+        {
+            RepoEvent::UnwantBlock(cid)
+        } else {
+            panic!("logic error: RepoEvent can only be created from a Request::GetBlock");
+        }
+    }
+}
```

Comment:
since this conversion can panic, I wonder if it should be `fn to_repo_event(self) -> Result<RepoEvent, Self>` instead? having the panic message from here would perhaps be more suitable where the conversion is needed inside `.expect(...)`?

Though, I haven't yet found where this is used.

Reply:
that panic is a logic error, though - it _should_ panic if it's used against its purpose

Reply:
though it is a `From` impl, so it makes sense that it shouldn't fail; I'll change it to a `TryFrom` and add an `unwrap` to the callsite.